lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java

```diff
@@ -95,15 +95,18 @@
 
     this.handlerContainer.setTransport(this.transport);
 
-    this.myIdentifier = new SocketRemoteIdentifier((InetSocketAddress)this.transport.getLocalAddress());
+    InetSocketAddress address = new InetSocketAddress(
+        localAddressProvider.getLocalAddress(),
+        this.transport.getListeningPort());
+    this.myIdentifier = new SocketRemoteIdentifier(address);
 
     this.reSendStage = new RemoteSenderStage(codec, this.transport, 10);
 
     StageManager.instance().register(this);
 
     final int counter = COUNTER.incrementAndGet();
 
-    LOG.log(Level.FINEST,
+    LOG.log(Level.INFO,
```

Comment:
>INFO [](start = 18, length = 4)

Change back to FINEST before sending pull request to apache/master. I left it here as it is helps me in debugging the containers feature.